extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -553,6 +554,33 @@ export class Bind {
     return value;
   }
 
+  /**
+   * Returns a copy of the global state for an expression, after waiting for its
+   * associated 'amp-state' element to finish fetching data. If there is no
+   * corresponding 'amp-state' element in the DOM, then reject.
+   *
+   * e.g. "foo.bar".
+   * @param {string} expr
+   * @return {!Promise<*>}
+   */
+  getStateAsync(expr) {
+    const stateKey = expr.split('.')[0];
```

Comment:
```suggestion
    const stateKey = /^[^.]*/.exec(expr)[0];
```

Reply:
Why is the regex better? Faster? 
Is it worth the readability trade?